### ISSUE TITLE
Latest Version Troubles

### ISSUE THREAD

**Issue by Mike-EEE** [Nethereum/NethereumBlazor#6]
Diving into the rabbit hole here. 😆

Looks like Blazor has updated its entire bits lately and this project's dependencies are out of date.

For instance, it seems that a new Blazor project from this resource utilizes `.razor` files whereas this project uses `.cshtml` files.

I am a total newb looking through this, but if it's a matter of updating file extensions and dependencies, I would be interested in submitting a PR to update everything. 👍

**Comment by juanfranblanco** [comment]
Wow that will be great!! Note that to integrate reactive UI this is not your vanilla Blazor.

**Comment by Mike-EEE** [comment]
OK COOL. 😁  I will see what I can do and update here if I can/can't. 👍

**Comment by Mike-EEE** [comment]
Sure enough, I have a nasty one that appears to be due to DynamicData's System.Reactive.

I see a `System.Reactive.dll` checked into source... is the idea to copy that in somehow?

Here's the stack I am seeing:

```
Uncaught (in promise) Error: System.TypeLoadException: Could not find method '.ctor' due to a type load error: Could not load type of field 'DynamicData.ObservableCache`2[TObject,TKey]:_changes' (0) due to: Could not resolve type with token 0100002b from typeref (expected class 'System.Reactive.Subjects.Subject`1' in assembly '') assembly: type:System.Reactive.Subjects.Subject`1 member:(null) assembly:DynamicData.dll type:ObservableCache`2 member:(null)
  at NethereumBlazor.Services.AccountsService..ctor (NethereumBlazor.Services.IWeb3ProviderService web3ProviderService) [0x0001a] in C:\- Remote\NethereumBlazor\NethereumBlazor\Services\AccountsService.cs:35 
  at NethereumBlazor.Startup.ConfigureServices (Microsoft.Extensions.DependencyInjection.IServiceCollection services) [0x00012] in C:\- Remote\NethereumBlazor\NethereumBlazor\Startup.cs:14 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) <0x2538fb0 + 0x000d2> in <e904f5829c7444acab57d4661a39f734>:0 
--- End of stack trace from previous location where exception was thrown ---

  at Microsoft.AspNetCore.Blazor.Hosting.ConventionBasedStartup.ConfigureServices (Microsoft.Extensions.DependencyInjection.IServiceCollection services) <0x2523480 + 0x0007e> in <91b564433c704544acd7894656c3aff9>:0 
  at (wrapper delegate-invoke) System.Action`1[Microsoft.Extensions.DependencyInjection.IServiceCollection].invoke_void_T(Microsoft.Extensions.DependencyInjection.IServiceCollection)
  at Microsoft.AspNetCore.Blazor.Hosting.WebAssemblyHostBuilderExtensions+<>c__DisplayClass1_0.<ConfigureServices>b__0 (Microsoft.AspNetCore.Blazor.Hosting.WebAssemblyHostBuilderContext context, Microsoft.Extensions.DependencyInjection.IServiceCollection collection) <0x2523020 + 0x00014> in <91b564433c704544acd7894656c3aff9>:0 
  at (wrapper delegate-invoke) System.Action`2[Microsoft.AspNetCore.Blazor.Hosting.WebAssemblyHostBuilderContext,Microsoft.Extensions.DependencyInjection.IServiceCollection].invoke_void_T1_T2(Microsoft.AspNetCore.Blazor.Hosting.WebAssemblyHostBuilderContext,Microsoft.Extensions.DependencyInjection.IServiceCollection)
  at Microsoft.AspNetCore.Blazor.Hosting.WebAssemblyHostBuilder.CreateServiceProvider () <0x24fffd0 + 0x000e8> in <91b564433c704544acd7894656c3aff9>:0 
  at Microsoft.AspNetCore.Blazor.Hosting.WebAssemblyHostBuilder.Build () <0x24ecb98 + 0x00050> in <91b564433c704544acd7894656c3aff9>:0 
  at NethereumBlazor.Program.Main (System.String[] args) [0x00001] in C:\- Remote\NethereumBlazor\NethereumBlazor\Program.cs:9 
    at Object.callMethod (http://localhost:52910/_framework/blazor.webassembly.js:1:35424)
    at Object.callEntryPoint (http://localhost:52910/_framework/blazor.webassembly.js:1:35002)
    at http://localhost:52910/_framework/blazor.webassembly.js:1:31090
    at http://localhost:52910/_framework/blazor.webassembly.js:1:29898
    at Object.next (http://localhost:52910/_framework/blazor.webassembly.js:1:30003)
    at i (http://localhost:52910/_framework/blazor.webassembly.js:1:28749)
```

**Comment by Mike-EEE** [comment]
doh... I created a pull request, but then I realized, that I didn't need to do that.  Oh well... it's there anyways. 😁  Still not proficient with git and its workflows yet.

**Comment by juanfranblanco** [comment]
Reactive UI is a fork and a submodule so you need to do ```git submodule init``` (the idea is that any changes that are made in the fork to support Blazor can be pushed eventually to ReactiveUI)

**Comment by Mike-EEE** [comment]
Ah ok... well the mystery deepens, then.  I do have ReactiveUI locally and everything compiles fine and launches.  It's only when trying to run the application that it throws that error at runtime.

I did run the above command, as well as `git submodule update --init --recursive` from this article, but that did not seem to fix [this issue](https://stackoverflow.com/questions/1030169/easy-way-to-pull-latest-of-all-git-submodules), unfortunately.

**Comment by juanfranblanco** [comment]
I modified ReactiveUI slightly did you used that version?

**Comment by Mike-EEE** [comment]
I am using what is in the repo. 😁

To assist, this reproduces the error:

```
	public class Program
	{
		public static void Main(string[] args)
		{
			new System.Reactive.Subjects.Subject<object>();
		}
	}
```

I am confused if we are talking about `ReactiveUI` or `System.Reactive`. 😆

**Comment by juanfranblanco** [comment]
hmm I might have been doing some experimental shit that got checked in by mistake :) Let me pull out master again and give it twirl.

**Comment by Mike-EEE** [comment]
Great, thank you.  To be sure this looks like a `System.Reactive` issue and not `ReactiveUI`.  

I do see a `System.Reactive.dll` in the root of NetereumBlazor.csproj.  Is that used or should we delete that?

**Comment by juanfranblanco** [comment]
Ok let me try to do the port as it is not as straight forward.. as there a few breaking changes on blazor, before I get to that part :)

**Comment by juanfranblanco** [comment]
@Mike-EEE  Done :) hopefully it is ok for you. 

I had some issue on startup on porting, mainly not happy to start processing the timer based services until fully completed the startup.

Other than that, changes.. etc

I have removed that System.Reactive.dll not needed

I still use the old Reactive.UI fork (that had no issues).  

I did a dotnet restore / dotnet build directly in the Reactive.UI project folder.

**Comment by Mike-EEE** [comment]
Woohoo!  It works!  Although, I have no idea why it was broken on my machine.  I do not see anything in your commit that is different from my pull request, except that you didn't upgrade `DynamicData` whereas I did.  I did try to revert that, but I still got the same error.

Sorry to make you go through that, that's _exactly_ that I tried to avoid. 😆  I will go ahead and close #8 and probably flatten my fork, too. :P
